File: spring-webflux-plain-java-example/src/test/java/de/cfranzen/examples/webflux/CustomerControllerTest.java
package de.cfranzen.examples.webflux;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.reactive.AutoConfigureWebTestClient;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.web.reactive.server.WebTestClient;

import static org.assertj.core.api.Assertions.assertThat;

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@AutoConfigureWebTestClient
class CustomerControllerTest {

    @Autowired
    private WebTestClient client;

    @Test
    void testCreationOfCustomer() {
        // Given
        final var request = new CustomerCreationRequest("Peter", "Pan");

        // When
        var result = client.post()
                .uri("/customer")
                .bodyValue(request)
                .exchange();

        // Then
        var customer = result
                .expectStatus()
                .isOk()
                .expectBody(Customer.class)
                .returnResult()
                .getResponseBody();

        assertThat(customer)
                .usingRecursiveComparison()
                .ignoringFields("id")
                .isEqualTo(new Customer("Peter", "Pan"));
        assertThat(customer.getId()).isNotNull();
    }
}

File: spring-webflux-kotlin-example/src/test/kotlin/de/cfranzen/examples/webflux/kotlin/CustomerControllerTest.kt
package de.cfranzen.examples.webflux.kotlin

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.webtestclient.autoconfigure.AutoConfigureWebTestClient
import org.springframework.test.web.reactive.server.WebTestClient

@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
@AutoConfigureWebTestClient
internal class CustomerControllerTest(
    @Autowired private val client: WebTestClient
) {

    @Test
    fun testCreationOfCustomer() {
        // Given
        val request = CustomerCreationRequest(firstName = "Peter", lastName = "Pan")

        // When
        val result = client
            .post()
            .uri("/customer")
            .bodyValue(request)
            .exchange()

        // Then
        val customer = result
            .expectStatus()
            .isOk()
            .expectBody(Customer::class.java)
            .returnResult()
            .responseBody!!

        assertThat(customer)
            .usingRecursiveComparison()
            .ignoringFields("id")
            .isEqualTo(Customer(firstName = "Peter", lastName = "Pan"))
        assertThat(customer.getId()).isNotNull()
    }
}